CHANGELOG.md

```diff
@@ -1,3 +1,81 @@
+## Unreleased
+
+### Breaking Change: Remove `-state` option
+
+If you use the implicit Local Backend by not having a Backend specified any of your stacks you need to migrate to using an explicit Local Backend. The new default for path will be the same as the now removed `-state` option.
+
+#### Typescript
+
+```ts
+import { LocalBackend } from "cdktf";
+import * as path from "path";
+
+class MyStack extends Stack {
+  constructor(scope: Construct, id: string) {
+    super(scope, id);
+
+    new LocalBackend(this);
```

Comment:
Previously we set the backends from the "outside" (e.g. in the templates when using Terraform Cloud). While both ways work and I have no especially hard feelings about it, my preference still lies with defining it from the outside. In my opinion this way we advocate for putting the configuration of the backends into the `App` instead of into each stack. If the stacks are moved to their own files at some point, so would the backend config along with them, "scattering" them throughout the project. But that is just personal preference and I'd like to get your and @skorfmann's views on this.

Reply:
Ah, makes sense to me. I prefer to have them at the top of my stack, but I can see your point. Shall I change it in general?

Reply:
@jsteinich What happens if you want to instantiate your stack multiple times? You could of course parameterize it somehow (i.e. based on the stack name, such as with the new default for the `path` of the LocalBackend), but I personally would rather configure it differently from outside of the stack (i.e. not in the constructor) to be explicit about where different instances of stacks go.

Reply:
I think either approach is equally valid.

I like how there is a logical replacement of backend happening in the hooks where previously there was just some code updating. If we do move it back to the app level, I'd like to see having the local backend still in the file so that we keep that replacement pattern.

Reply:
Thanks to https://github.com/hashicorp/terraform-cdk/blob/84dfc92b27f657a38c960ca4f40c84b64ff2039b/packages/cdktf/lib/terraform-stack.ts#L196 Users won't need to do any extra work here 👍